File: lib/oberon/base/math.lua
-------------------------------------------------------------------------------
-- Mathematical Functions.
-- This library is an interface to the standard C math library.
-- It provides all its functions inside the table math.
-- @module math

math = require 'math'

-------------------------------------------------------------------------------
-- Returns round number with precision.
-- @function [parent=#math] round
-- @param #number number
-- @param #number precision
-- @return #number

function math.round(number, precision)
  return math.floor((number + (0.5/(10 ^ precision))) * (10 ^ precision)) / (10 ^ precision)
end

-------------------------------------------------------------------------------
-- Returns round number truncate with precision.
-- @function [parent=#math] truncate
-- @param #number number
-- @param #number precision
-- @return #number

function math.truncate(number, precision)
  return math.floor(number * (10 ^ precision)) / (10 ^ precision)
end

return math

File: src/oberon/base/math.cpp
#include <cmath>
#include <oberon/helper>

int math::abs(int number)
{
  return std::abs(number);
}

double math::acos(double number)
{
  return std::acos(number);
}

double math::asin(double number)
{
  return std::asin(number);
}

double math::atan(double number)
{
  return std::atan(number);
}

double math::atan2(double x, double y)
{
  return std::atan2(x, y);
}

double math::ceil(double number)
{
  return std::ceil(number);
}

double math::cos(double number)
{
  return std::cos(number);
}

double math::cosh(double number)
{
  return std::cosh(number);
}

double math::floor(double number)
{
  return std::floor(number);
}

double math::round(double number, int precision)
{
  double dprecision = std::pow(10, precision);
  return std::floor((number + (0.5/(dprecision))) * (dprecision)) / dprecision;
}

double math::truncate(double number, int precision)
{
  double dprecision = std::pow(10, precision);
  return std::floor(number * (dprecision)) / dprecision;
}
